lib/rig_inbound_gateway_web/vconnection.ex

```diff
@@ -0,0 +1,204 @@
+defmodule RigInboundGatewayWeb.VConnection do
+  @moduledoc """
+  The VConnection acts as an abstraction for the actual connections.
+
+  It monitors the processes, sends out heartbeats and subscription refreshes and sends out events to the serverless engine when intialized, connection is lost or connection is re-established.
+  """
+
+  alias Rig.EventFilter
+  alias Rig.RingBuffer
+  alias RigCloudEvents.CloudEvent
+  alias RigInboundGateway.Events
+
+  require Logger
+
+  use GenServer
+
+  # TODO: Discuss a reasonable kill delay
```

Comment:
should be 10-15mins imho

Reply:
That long? Isn't that going to use many resources?

Reply:
true. what about 5 mins then? could be a good compromise. given that on a regular connection a user should see something around 5 events every 10 seconds, the event buffer size should be set 150 to reflect this.

Reply:
5 mins sounds like a reasonable default. Shouldn't we offer the option to configure this tho?

Reply:
yes, good idea. What about `IDLE_CONNECTION_TIMEOUT`?